packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -150,21 +356,64 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
     this.match = options.match;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const pathRewriteConfig = this.match?.pathRewrite?.bind(scope);
+    const pathMatchConfig = this.match?.path?.bind(scope);
+    const defaultTargetHostname = this.match?.rewriteRequestHostname;
+
+    // Set prefix Match to '/' if none on match properties are defined.
+    const prefixPathMatch = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : pathMatchConfig?.prefixMatch;
+    const pathMatch = pathMatchConfig?.pathMatch;
+    const prefixPathRewrite = pathRewriteConfig?.prefixPath;
+    const pathRewrite = pathRewriteConfig?.path;
+
+    // Checks if the specified values are starting with '/'.
+    validateStartWith(prefixPathMatch, pathMatch?.exact, prefixPathRewrite?.value, pathRewrite?.exact);
+
+    if (prefixPathRewrite && !prefixPathMatch) {
+      throw new Error('HTTP Gateway Route prefix match must be provided if a prefix rewrite was provided.');
+    }
+    if (pathRewrite && prefixPathMatch) {
+      throw new Error('HTTP Gateway Route prefix match and path rewrite cannot both be provided.');
     }
+
     const httpConfig: CfnGatewayRoute.HttpGatewayRouteProperty = {
       match: {
-        prefix: prefixPath,
+        prefix: prefixPathMatch,
+        path: pathMatch,
+        hostname: this.match?.hostname?.bind(scope).hostnameMatch,
+        method: this.match?.method,
+        headers: this.match?.headers?.map(header => header.bind(scope).headerMatch),
+        queryParameters: this.match?.queryParameters?.map(queryParameter => queryParameter.bind(scope).queryParameter),
       },
       action: {
         target: {
           virtualService: {
             virtualServiceName: this.routeTarget.virtualServiceName,
           },
         },
+        rewrite: defaultTargetHostname === false || pathRewriteConfig
+          ? {
+            hostname: defaultTargetHostname === false
+              ? {
+                defaultTargetHostname: 'DISABLED',
+              }
+              : undefined,
```

Comment:
```suggestion
            hostname: defaultTargetHostname === undefined
              ? undefined
              : {
                defaultTargetHostname: defaultTargetHostname? 'ENABLED' : 'DISABLED',
              },
```

Reply:
Same with [this comment.](https://github.com/aws/aws-cdk/pull/15395#discussion_r665772288)

Reply:
I would still do it, I don't see much advantage in removing the `'ENABLED'` option.

Reply:
Addressed